crates/re_types_builder/src/lib.rs

```diff
@@ -365,9 +379,41 @@ pub fn generate_cpp_code(
     arrow_registry: &ArrowRegistry,
 ) {
     re_tracing::profile_function!();
+
+    // 1. Generate code files.
     let mut gen = CppCodeGenerator::new(output_path.as_ref());
-    let filepaths = gen.generate(reporter, objects, arrow_registry);
-    generate_gitattributes_for_generated_files(&output_path, filepaths.into_iter());
+    let mut files = gen.generate(reporter, objects, arrow_registry);
+    // 2. Generate attribute files.
+    generate_gitattributes_for_generated_files(&mut files);
+    // 3. Write all files.
+    {
+        use rayon::prelude::*;
+
+        re_tracing::profile_scope!("write_files");
+
+        files.par_iter().for_each(|(filepath, contents)| {
+            // There's more than cpp/hpp files in here, don't run clang-format on them!
+            let contents =
+                if filepath.extension() == Some("cpp") || filepath.extension() == Some("hpp") {
```

Comment:
style choice, but this is nice and short:

```suggestion
                if matches!(filepath.extension(), Some("cpp" | "hpp")) {
```